flang/lib/Lower/Bridge.cpp

```diff
@@ -1621,6 +1621,8 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     }
     if (const auto *details =
             sym.detailsIf<Fortran::semantics::ObjectEntityDetails>()) {
+      // FIXME: an exported module variable will have external linkage.
```

Comment:
I think what's happening here is that all variables are being given internal linkage, which is fine for f77.  The fixme means that this code will need to be updated for F90+ because module variables have external linkage.

Reply:
+1